metapool/sample_sheet.py

```diff
@@ -819,6 +852,63 @@ def _validate_sample_sheet_metadata(self, metadata):
         return msgs
 
 
+class KLSampleSheetWithSampleContext(KLSampleSheet):
+    COLS_BY_EXTRA_DF_SECTION_NAME = {
+        BIOINFORMATICS_KEY: _BASE_BIOINFORMATICS_COLS,
+        CONTACT_KEY: _CONTACT_COLS,
+        SAMPLE_CONTEXT_KEY: _SAMPLE_CONTEXT_COLS
+    }
+
+    _ALL_METADATA = KLSampleSheet._ALL_METADATA.copy()
+    _ALL_METADATA[SAMPLE_CONTEXT_KEY] = None
+
+    sections = ('Header', 'Reads', 'Settings', 'Data', BIOINFORMATICS_KEY,
+                CONTACT_KEY, SAMPLE_CONTEXT_KEY)
+
+    def __new__(cls, path=None, *args, **kwargs):
+        """
+            Override new() so that base class cannot be instantiated.
+        """
+        if cls is KLSampleSheetWithSampleContext:
+            raise TypeError(
+                f"only children of '{cls.__name__}' may be instantiated")
+
+        instance = super(KLSampleSheetWithSampleContext, cls).__new__(
+            cls, *args, **kwargs)
+        return instance
+
+    def __init__(self, path=None):
+        """Knight Lab's SampleSheet subclass that includes SampleContext
+
+        Expands Knight Lab SampleSheet to include a new (required) section
+        called 'SampleContext'. This section is used to store information
+        about the blanks and other controls used in the sequencing run.
+
+        Parameters
+        ----------
+        path: str, optional
+            File path to the sample sheet to load.
+        """
+
+        # NB: it matters that this come *before* the __init__ call;
+        # __init__ calls _parse, which will automatically populate the
+        # SampleContext section if it is present in the file--but only if
+        # it is defined here first.
+        self.SampleContext = None
+        super().__init__(path=path)
+
+        if self.path:
+            pass
+
+            # TODO: do we need to normalize booleans for SampleContext?
```

Comment:
I have done this for all the dataframe sections ... I'm going to leave doing it for arbitrary attributes for the future, since I don't see any of them right now.